server/api/dataset.py

```diff
@@ -48,3 +50,51 @@ def find(self, params):
     def getDataset(self, dataset, params):
         return self.model('dataset', 'isic_archive').filter(
             dataset, self.getCurrentUser())
+
+    @describeRoute(
+        Description('Create a lesion image dataset.')
+        .param('uploadFolderId', 'The ID of the folder that contains images '
+               'and metadata.')
+        .param('name', 'Name of the dataset.')
+        .param('description', 'Description of the dataset.', required=False)
+        .param('license', 'License of the dataset.', required=False)
+        .param('signature', 'Signature of license agreement.', required=True)
+        .param('anonymous', 'Whether to use an anonymous attribution for the '
+               'dataset', dataType='boolean', required=False)
+        .param('attribution', 'Attribution of the dataset.', required=False)
+    )
+    @access.user
+    def createDataset(self, params):
+        self.requireParams(('uploadFolderId', 'name'), params)
+
+        user = self.getCurrentUser()
```

Comment:
We should also add a check to make sure that the user is an authorized uploader. A magic group is probably the easiest way for now.

We also need to add this check when rendering the "Upload dataset" button / route on the client.